great_expectations/datasource/data_connector/util.py

```diff
@@ -342,29 +345,41 @@ def sanitize_prefix_for_s3(text: str) -> str:
 
 
 def normalize_directory_path(
-    dir_path: str, root_directory_path: Optional[str] = None
-) -> str:
+    dir_path: Union[pathlib.Path, str],
+    root_directory_path: Optional[Union[pathlib.Path, str]] = None,
+) -> pathlib.Path:
+    if isinstance(dir_path, str):
+        dir_path = pathlib.Path(dir_path)
+
+    if isinstance(root_directory_path, str):
+        root_directory_path = pathlib.Path(root_directory_path)
+
     # If directory is a relative path, interpret it as relative to the root directory.
-    if Path(dir_path).is_absolute() or root_directory_path is None:
+    if dir_path.is_absolute() or root_directory_path is None:
         return dir_path
 
-    return str(Path(root_directory_path).joinpath(dir_path))
+    return root_directory_path.joinpath(dir_path)
 
 
 def get_filesystem_one_level_directory_glob_path_list(
-    base_directory_path: str, glob_directive: str
+    base_directory_path: Union[pathlib.Path, str], glob_directive: str
```

Comment:
Same note about old-style annotations and `PathStr`

Reply:
@cdkini Thanks -- comitted.